cfme/physical/physical_switch.py

```diff
@@ -0,0 +1,238 @@
+            return self.db_id
+        else:
+            return self.db_id
+
+    def wait_to_appear(self):
+        """Waits for the Switch to appear in the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the Switch to appear...")
+        wait_for(
+            lambda: self.exists,
+            message="Wait for the Switch to appear",
+            num_sec=1000,
+            fail_func=view.browser.refresh
+        )
+
+    def wait_for_delete(self):
+        """Waits for the Switch to remove from the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the Switch to delete...")
+        wait_for(
+            lambda: not self.exists,
+            message="Wait for the Switch to disappear",
+            num_sec=500,
+            fail_func=view.browser.refresh
+        )
+
+
+@attr.s
+class PhysicalSwitchCollection(BaseCollection):
+    """Collection object for the :py:class:`cfme.infrastructure.host.PhysicalSwitch`."""
+
+    ENTITY = PhysicalSwitch
+
+    def select_entity_rows(self, physical_switches):
+        """ Select all physical Switch objects """
+        physical_switches = list(physical_switches)
+        checked_physical_switches = list()
+        view = navigate_to(self, 'All')
+
+        for physical_switch in physical_switches:
+            view.entities.get_entity(name=physical_switch.name, surf_pages=True).check()
+            checked_physical_switches.append(physical_switch)
+        return view
+
+    def all(self, provider):
+        """returning all physical_switches objects"""
+        physical_switch_table = self.appliance.db.client['switches']
+        ems_table = self.appliance.db.client['ext_management_systems']
+        physical_switch_query = (
+            self.appliance.db.client.session
+                .query(physical_switch_table.name, ems_table.name)
+                .join(ems_table, physical_switch_table.ems_id == ems_table.id))
+        provider = None
+
+        if self.filters.get('provider'):
+            provider = self.filters.get('provider')
+            physical_switch_query = physical_switch_query.filter(ems_table.name == provider.name)
+        physical_switches = []
+        for name, ems_name in physical_switch_query.all():
+            physical_switches.append(self.instantiate(name=name,
+                                    provider=provider or get_crud_by_name(ems_name)))
+        return physical_switches
+
+    def find_by(self, provider, ph_name):
+        """returning all physical_switches objects"""
+        physical_switch_table = self.appliance.db.client['switches']
+        ems_table = self.appliance.db.client['ext_management_systems']
+        physical_switch_query = (
+            self.appliance.db.client.session
+                .query(physical_switch_table.name, ems_table.name)
+                .join(ems_table, physical_switch_table.ems_id == ems_table.id))
+        provider = None
```

Comment:
**REQUIRED**
This line should be removed.

Reply:
Done. This is the diff for the next method.

Reply:
this was not corrected

Reply:
Hi @rlbabyuk. I removed from the `find_by` function (around [line 212]( https://github.com/ManageIQ/integration_tests/pull/7422/files#diff-b209a1a0c0a9470efd617279e1e7b5c8R212)). This is in a similar block at `all` function and I use it at line 201. Should I still remove it?